netlogo-gui/src/main/ide/JumpToDeclaration.scala

```diff
@@ -0,0 +1,88 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.ide
+
+import org.nlogo.core.{Femto, Token, TokenType, TokenizerInterface}
+import org.nlogo.editor.EditorArea
+
+
+object JumpToDeclaration {
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  def jumpToDeclaration(cursorPosition: Int, editorArea: EditorArea): Unit ={
+    val tokenOption = findTokenContainingPosition(editorArea.getText(), cursorPosition)
+    for(token <- tokenOption) {
+      val t = getDeclaration(token, editorArea.getText)
+      editorArea.select(t.start, t.end)
+    }
+  }
+
+  def getDeclaration(token: Token, source: String): Token = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source).toSeq
+    val (globals, owns) = parseOwns(source)
+    val indexOfToken = iterator.indexWhere(t => t.start == token.start)
+    var stack = 0
+    for(i <- indexOfToken to 0 by -1) {
+      val tok = iterator(i)
+      if (tok.tpe.equals(TokenType.CloseBracket)) {
+        stack -= 1
+      }
+      if (tok.tpe.equals(TokenType.OpenBracket)) {
+        stack += 1
+      }
+      if(stack == 0 && tok.text.equalsIgnoreCase(token.text) && iterator(i - 1).text.equalsIgnoreCase("let")) {
```

Comment:
I have tested this on the following model 
```
globals [
  scared
]

me-own[
  something
]
you-own [
  something
]

to newF
  scared
  let scared
  scared
END
to new [scared]
  scared
  [
    let scared
    scared
    [
      scared
      let scared
      scared
      something
    ]
  ]
  scared
end
```
For this model it works everywhere.